cmd/rekor-server/app/flags_test.go

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestLogRanges_Set(t *testing.T) {
+	tests := []struct {
+		name    string
+		arg     string
+		want    []LogRange
+		wantErr bool
+	}{
+		{
+			name: "one",
+			arg:  "1234=10",
+			want: []LogRange{
+				{
+					TreeId:     1234,
+					TreeLength: 10,
+				},
+			},
+		},
+		{
+			name: "two",
+			arg:  "1234=10,7234=17",
+			want: []LogRange{
+				{
+					TreeId:     1234,
+					TreeLength: 10,
+				},
+				{
+					TreeId:     7234,
+					TreeLength: 17,
+				},
+			},
+		},
+		{
+			name:    "invalid",
+			arg:     "1234=10,7234-17",
+			want:    []LogRange{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			l := &LogRanges{}
+			if err := l.Set(tt.arg); (err != nil) != tt.wantErr {
+				t.Errorf("LogRanges.Set() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if diff := cmp.Diff(tt.want, l.Ranges); diff != "" {
+				t.Errorf(diff)
+			}
+		})
+	}
+}
+
+func TestLogRanges_ResolveVirtualIndex(t *testing.T) {
+	lrs := LogRanges{
+		Ranges: []LogRange{
+			{TreeId: 1, TreeLength: 17},
+			{TreeId: 2, TreeLength: 1},
+			{TreeId: 3, TreeLength: 100},
+			{TreeId: 4},
```

Comment:
This makes sense here (for all new additions to go into `TreeId: 4`), but should we extend the command line arg to have a trailing value to match this? If the server panics and restarts, we don't want it to create a new tree but if the trillian instance reports back more than one tree that isn't specified in the list we wouldn't be able to disambiguate which one to use.

Reply:
+1, I thought about this but skipped it. We should. I'll fix it up here.